assets/js/search-categories-metabox.js

```diff
@@ -0,0 +1,186 @@
+		var selectedCategories = getSelectedCategories();
+
+		// Function definitions.
+
+		/**
+		 * @summary Sorts the Categories by name and then by checked items.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @global type selectedCategories List of selected Categories ID.
+		 */
+		function sortCategoriesListBySelectedItems() {
+
+			var selectedCategories = [],
+				otherCategories    = [],
+				checkBoxState;
+
+			$( 'ul#scm-categories-ul li' ).each(function () {
+				checkBoxState = $( this ).find( 'input' ).prop( 'checked' );
+				if ( true === checkBoxState ) {
+					selectedCategories.push( $( this ) );
+				} else {
+					otherCategories.push( $( this ) )
+				}
+			});
+
+			$( 'ul#scm-categories-ul' )
+				.html( selectedCategories )
+				.append( otherCategories );
+		}
+
+		/**
+		 * @summary Gets the User selected Categories.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @return array List of User selected Categories.
+		 */
+		function getSelectedCategories() {
+			// Category ID's.
+			var categories = [];
+			// Get selected post Category ID's by looping through selected checkboxes.
+			$.each( $( 'input.scm-category-list-item[type=checkbox]:checked' ), function() {
+				categories.push( $( this ).val() );
+			});
+			// Return the array of selected Category ID's.
+			return categories;
+		}
+
+		// Page Load defaults.
+		$( 'p#scm-category-add' ).hide();
+
+		// Event definitions.
+
+		// Sort Categories list so that the selected items appear at the top.
+		sortCategoriesListBySelectedItems();
+
+		// Capture the user selected Categories after page loads.
+		$( '#scm-categories-ul' ).on('click', '.scm-category-list-item', function() {
+
+			var checkBoxState = $( this ).prop( 'checked' );
+			var checkBoxVal   = $( this ).val();
+
+			/*
+			 * When user deselects, remove the item from
+			 * global selectedCategories
+			 */
+			if ( false === checkBoxState ) {
+				if ( selectedCategories.length > 0 && $.inArray( checkBoxVal, selectedCategories ) !== -1 ) {
+					selectedCategories = $.grep(selectedCategories, function( a ) {
+						return a !== checkBoxVal;
+					});
+				}
+				$( this ).removeAttr( 'checked' );
+			} else {
+				/*
+				 * When user selects, add the item to
+				 * global selectedCategories
+				 */
+				if ( selectedCategories.length === 0 || $.inArray( checkBoxVal, selectedCategories ) === -1 ) {
+					selectedCategories.push( checkBoxVal );
+				}
+				$( this ).attr('checked', 'checked');
+			}
+		});
+
+		/**
+		 * @summary Displays error message in Search Categories Metabox.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @global array selectedCategories List of selected Categories ID.
+		 */
+		function displayError() {
+			var errorMessage;
+
+			if ( $.type( ajax_object.something_wrong_error_message ) != 'undefined' ) {
+				errorMessage = ajax_object.something_wrong_error_message;
+			} else {
+				errorMessage = 'Something went wrong.';
+			}
+
+			$( 'ul#scm-categories-ul' ).html( '<li>' + errorMessage  + '</li>' );
```

Comment:
This could result in a security issue if someone hijacks your AJAX request.

Eg: What happens if they change the message to a script tag?

Ideally you should just use `text` method or create the element and then append them.